sidecar/init.go

```diff
@@ -44,6 +50,61 @@ func NewInitCommand(cfg *Config) *cobra.Command {
 	return cmd
 }
 
+// checke leader or service
+func CheckServiceExist(cfg *Config, service string) bool {
+	serviceURL := fmt.Sprintf("http://%s-%s:%v%s", cfg.ClusterName, service, utils.XBackupPort, "/health")
+	req, err := http.NewRequest("GET", serviceURL, nil)
+	if err != nil {
+		log.Info("failed to check available service", "service", serviceURL, "error", err)
+		return false
+	}
+
+	client := &http.Client{}
+	client.Transport = transportWithTimeout(serverConnectTimeout)
+	resp, err := client.Do(req)
+	if err != nil {
+		log.Info("service was not available", "service", serviceURL, "error", err)
+		return false
+	}
+
+	defer func() { _ = resp.Body.Close() }()
+
+	if resp.StatusCode != 200 {
+		log.Info("service not available", "service", serviceURL, "HTTP status code", resp.StatusCode)
+		return false
+	}
+
+	return true
+}
+
+// clone from leader or follower
+func runCloneAndInit(cfg *Config) error {
+	//check leader is exist?
+	serviceURL := ""
+	if len(serviceURL) == 0 && CheckServiceExist(cfg, "leader") {
+		serviceURL = fmt.Sprintf("http://%s-%s:%v", cfg.ClusterName, "leader", utils.XBackupPort)
```

Comment:
we can use `GetNameForResource`

Reply:
cannot， `(c *MysqlCluster) GetNameForResource ` need define mysqlcluster, here just is sidecar